### PR TITLE
Fix/standalone pump fix

### PR DESCRIPTION
**Pull request by Himorask**
Changes to make standalone pumps turn on for experiments.

### CODE REVIEW

**Comment by dieris** [comment]
Are you sure you need [this](https://github.com/BBN-Q/Auspex/pull/399/commits/ae53f77b3997e5392fcf913c31928057b3d2caf0)? I thought the key was just adding an `output` field to the `Source` object

**Comment by Himorask** [comment]
Hmm, maybe I misunderstood the changes we made... I'll take another look

File: src/auspex/instruments/yokogawa.py [hunk 1]
```diff
@@ -21,19 +22,20 @@ class YokogawaGS200(SCPIInstrument):
...
     protection_volts   = FloatCommand(scpi_string=":source:protection:voltage")
     protection_current = FloatCommand(scpi_string=":source:protection:current")
-    sense              = BoolCommand(scpi_string=":sense:state", value_map={True: "1", False: "0"})
-    output             = BoolCommand(scpi_string=":output:state", value_map={True: "1", False: "0"})
+    sense              = StringCommand(scpi_string=":sense:state", value_map={True: "1", False: "0"})
+    output             = StringCommand(scpi_string=":output:state", value_map={True: "1", False: "0"})
     sense_value        = FloatCommand(get_string=":fetch?")
     averaging_nplc     = IntCommand(scpi_string=":sense:nplc") # Number of power level cycles (60Hz)
-    ramp               = RampCommand(increment=1e-4, pause=20e-3, scpi_string=":source:level", value_range=(-100e-3,100e-3))
+ #   ramp               = RampCommand(increment=1e-4, pause=20e-3, scpi_string=":source:level", value_range=(-100e-3,100e-3))
```
Comment (grahamrow) [inline src/auspex/instruments/yokogawa.py]: Why get rid of this one?

File: src/auspex/experiment.py [hunk 2]
```diff
@@ -513,6 +513,14 @@ def run_sweeps(self):
             for w in wrs:
                 w.filename.value = inc_filename
         self.filenames = [w.filename.value for w in self.writers]
+        # Save ChannelLibrary version
```
Comment (grahamrow) [inline src/auspex/experiment.py]: I don't think I agree with this approach, at least not unilaterally. User should have the option of not applying changes, for example. If somebody runs an experiment in a loop this is going to create an enormous database.

File: src/auspex/instruments/agilent.py [hunk 3]
```diff
@@ -828,7 +828,7 @@ def get_trace(self, measurement=None):
         self.interface.write(":CALCulate:PARameter:SELect '{}'".format(measurement))
         self.reaverage()
         self.interface.write(":CALC:DATA? SDATA")
-        block =  self.interface.read_raw(size=256)
+        block =  self.interface.read_raw()
```
Comment (grahamrow) [inline src/auspex/instruments/agilent.py]: We need to test this on other machines. @gribeill and I have had a miserable time getting VNA transfers working without a chunk limit on certain systems.

File: src/auspex/instruments/interface.py [hunk 4]
```diff
@@ -69,6 +69,7 @@ def ESE(self):
     def ESR(self):
         return self._resource.query("*ESR?") # Standard Event Status Register Query
     def IDN(self):
+        import pdb;pdb.set_trace()
```
Comment (grahamrow) [inline src/auspex/instruments/interface.py]: ✂️

File: src/auspex/instruments/yokogawa.py (hunk 1 above)
Comment (dieris) [reply src/auspex/instruments/yokogawa.py]: This PR had a lot more than it says :) The only intended commit was the one on standalone instruments, but to be verified by @Himorask .

File: src/auspex/experiment.py (hunk 2 above)
Comment (dieris) [reply src/auspex/experiment.py]: Fair enough, I agree it should be optional. This was already merged to develop, I'll change it from there

**Comment by Himorask** [comment]
Hmm, I had tried to only commit one change but it seems it ended up capturing a lot more than that

**Comment by grahamrow** [comment]
What's the status here?

**Comment by dieris** [comment]
I'll make a new PR for standalone sources only

**Pull request closed without merging**